syntax = "proto2";

package lego.service.protobuf;

message GetVpnInfoRequest {
    optional bytes pubkey = 1;
    optional bytes sign_challenge = 2;
    optional bytes sign_response = 3;
    optional bool heartbeat = 4;
}

message GetVpnInfoResponse {
	optional bytes ip = 1;
	optional uint32 svr_port = 2;
	optional uint32 route_port = 3;
    optional bytes country = 4;
    optional int32 secnum = 5;
    optional bytes pubkey = 6;
}

message ServiceMessage {
	optional GetVpnInfoRequest vpn_req = 1;
	optional GetVpnInfoResponse vpn_res = 2;
}